sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -467,8 +467,9 @@ private bool TryGetReturnTypePropertyBindings(ITypeSymbol returnTypeSymbol, bool
                         return false;
                     }
 
-                    // Check if this attribute is an HttpResponseData type attribute
-                    if (prop is IPropertySymbol property && SymbolEqualityComparer.Default.Equals(property.Type, _knownFunctionMetadataTypes.HttpResponse))
```

Comment:
Previously it worked to have an output type of `HttpResponse`.  Would this require existing `HttpResponse` response properties to start using `HttpResponseOutputAtribute` making this a breaking change?

Reply:
We're adding this new case as part of an "OR" statement, so previous scenarios should still work - for example, `HttpResponseData` would still not require the attribute. I don't believe the output type of `HttpResponse` in this scenario was not a documented approach from our end, but I'll make a note to test it.

Reply:
It wasn't documented but I was just in the process of process of migrating from in-proc function to isolated and ran into #2205 (exception `IFeatureCollection has been disposed.` when returning the `HttpResponseData`) as we typically use IActionResult response using `Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore` for http. I switched to use HttpResponse on the output object to fix the issue.  We're currently returning HttpResponse now which is working, at least until we update to this version.  Is there any way to provide an Analyzer or something that would warn on upgrade that this is necessary?

Reply:
@dandenton Good callout - tracking this issue [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/2352). 

CC: @fabiocav @mattchenderson

Reply:
I'm tempted to say this should be within the scope of the changes.

Reply:
Or at the very least, let's get this going in parallel